modules/apps/commerce/commerce-product-tax-category-web/src/main/java/com/liferay/commerce/product/tax/category/web/internal/display/context/CPTaxCategoryDisplayContext.java

```diff
@@ -172,14 +173,26 @@ _renderRequest, getPortletURL(), null,
 			_getCPTaxCategoryOrderByComparator(
 				getOrderByCol(), getOrderByType()));
 		_searchContainer.setOrderByType(getOrderByType());
-		_searchContainer.setResultsAndTotal(
-			() -> _cpTaxCategoryService.getCPTaxCategories(
-				themeDisplay.getCompanyId(), _searchContainer.getStart(),
-				_searchContainer.getEnd(),
-				_searchContainer.getOrderByComparator()),
-			_cpTaxCategoryService.getCPTaxCategoriesCount(
-				themeDisplay.getCompanyId()));
-		_searchContainer.setRowChecker(_getRowChecker());
+
+		if (!Validator.isBlank(getKeywords())) {
+			_searchContainer.setResultsAndTotal(
```

Comment:
why don't we just use the search method instead of introducing this if?

Reply:
done